packages/augur-core/source/contracts/trading/ShareToken.sol

```diff
@@ -30,30 +31,39 @@ contract ShareToken is ITyped, Initializable, VariableSupplyToken, IShareToken {
     address public claimTradingProceeds;
     IProfitLoss public profitLoss;
 
-    bool private shouldUpdatePL;
-
-    modifier doesNotUpdatePL() {
-        shouldUpdatePL = false;
-        _;
-        shouldUpdatePL = true;
-    }
+    mapping(address => bool) private doesNotUpdatePnl;
 
     function initialize(IAugur _augur, IMarket _market, uint256 _outcome, address _erc1820RegistryAddress) external beforeInitialized {
         endInitialization();
         market = _market;
+        universe = _market.getUniverse();
         outcome = _outcome;
         augur = _augur;
-        shouldUpdatePL = true;
-        createOrder = _augur.lookup("CreateOrder");
-        fillOrder = _augur.lookup("FillOrder");
-        cancelOrder = _augur.lookup("CancelOrder");
+
+        address _createOrder = _augur.lookup("CreateOrder");
+        address _fillOrder = _augur.lookup("FillOrder");
+        address _cancelOrder = _augur.lookup("CancelOrder");
+
+        doesNotUpdatePnl[_createOrder] = true;
+        doesNotUpdatePnl[_fillOrder] = true;
+        doesNotUpdatePnl[_cancelOrder] = true;
+        
+        createOrder = _createOrder;
+        fillOrder = _fillOrder;
+        cancelOrder = _cancelOrder;
+
         completeSets = _augur.lookup("CompleteSets");
         claimTradingProceeds = _augur.lookup("ClaimTradingProceeds");
         profitLoss = IProfitLoss(_augur.lookup("ProfitLoss"));
         erc1820Registry = IERC1820Registry(_erc1820RegistryAddress);
         initialize1820InterfaceImplementations();
     }
 
+    function setUniverse(IUniverse _universe) external {
```

Comment:
This won't work for a fork. Also this won't work for the model of 1155 coming up soon anyway so probably should just remove. (We could keep a mapping of market to universe for the 1155 model perhaps but not sure its worth it there)

Reply:
What about it won't work? This gets called during migrateThroughOneFork to update it to the new universe, so it should have the correct value at any given time

Reply:
Ah I see. Can stay for this PR then though we'll need to rethink with 1155 as noted